### PR TITLE
Luizoamorim/from ilyas browser fixes 2

### PR DESCRIPTION
**Pull request by luizoamorim**
In this PR I'll be fixing some issues from here : [https://github.com/EYBlockchain/nightfall_3_private/issues/74](https://github.com/EYBlockchain/nightfall_3_private/issues/74) for browser.

# Browser issues

- [x] **Issue 1 - If you put an ethereum address instead of a pkd L2 address then you get some error in the console and the modal of generating the proof stays forever**

Do a verification using the difference of length between l1 hash and l2 hash.

- [x] **Issue 4 -  L2 Bridge page - transact MAX - Doesn't work - console error: Uncaught TypeError: Cannot set properties of null (setting 'value')**

- [x] **Issue 5 - Deposits - can create and send transactions for amounts bigger than my current balance. Can create the commitment successfully, but when sending the tx Metamask will console.log an RPC error - the user won't notice (image attached in comments below). You end up with an 'invalid' commitment.**

I think the solution is block this kind of transactions in frontend.

- [x] **Issue 6 - Deposits - can create and send transactions with no value, however the process seems to 'complete' not causing side effects - hence the low priority**

### What I saw: 

- [x] *For send transaction is not being verified the address neither the ammount*

- [x] *For withdraw if you input a ammount greater than your balance the modal is opening*

- [x] *For deposit the modal is not opening for 0 ammount or ammount greater than balance, but is not showing any warning.*

- [x] **Issue 7 - E.g. I deposit 1 MATIC, after completing the tx, form is still 'set' to 1 MATIC. Suggestion - re-route users to Transactions page if tx is successful.**

- [x] **Issue 9 - Restrictions in deposits now it shows a message in red under the amount to warn the user but the Transfer button is not disabled, so I can continue with the deposit of a restricted amount.**

- [x] **Issue 10 - In transactions, clicking on arrow takes you to goerli etherscan. In mainnet wallet, this should be mainnet etherscan**

### CODE REVIEW

**Comment by luizoamorim** [comment]
The commit #e3eabc9713d97df9389d55df555bb4705713f936 is fixing the issue number 4. This way the button MAX back to work.

**Comment by luizoamorim** [comment]
## Issue 6
The commit 40688d1a8794bbed4a18f645bba7dd7726a06971 is fixing the problems for the deposit and withdraws transfers verification.

**Comment by luizoamorim** [comment]
## Issue 1 and the rest of 6

The last commits have the changes to fix the problems in send modal:
 - Verification if the address is in correct format.
 -  Verification if the user put a balance greater than zero or nor greater than its balance.

**Comment by luizoamorim** [comment]
## Issue 9 
In commit 40688d1a8794bbed4a18f645bba7dd7726a06971 line 329 I added the restriction for deposit. 
**I did put 0.25 hardcoded, maybe we can think in have it like some env var.**

line 329:
```
if (txType === 'deposit' &&
        new BigFloat(transferValue, token.decimals).toBigInt() > new BigFloat('0.25 ', token.decimals).toBigInt()
    ) {
      toast.error("Input value can't be greater than 0.25");
      return;
}
```

**Comment by luizoamorim** [comment]
## Issue 7
Was added a new feat in the last commits. If the subimitTx function return true, the user is redirected for transactions page.

File: wallet/src/components/BridgeComponent/index.jsx [hunk 1]
```diff
@@ -205,8 +212,10 @@ const BridgeComponent = () => {
               throw new Error(err);
             });
           break;
-        default:
+        }
+        default: {
```
Comment (IlyasRidhuan) [inline wallet/src/components/BridgeComponent/index.jsx:216]: Remove curly brackets here

File: wallet/src/components/BridgeComponent/index.jsx [hunk 2]
```diff
@@ -311,10 +320,30 @@ const BridgeComponent = () => {
...
       toast.error("Input value can't be greater than balance!");
-    else if (!transferValue) toast.warn('Input a value for transfer, please.');
-    else if (transferValue === 0) toast.warn("Input a value can't be zero.");
+      return;
+    }
+
+    if (
+      txType === 'deposit' &&
+      new BigFloat(transferValue, token.decimals).toBigInt() >
+        new BigFloat('0.25', token.decimals).toBigInt()
```
[the remark this answers is not in the document]
Comment (luizoamorim) [reply wallet/src/components/BridgeComponent/index.jsx:331]: Ah good to know! Tks! I'll change this.

File: wallet/src/components/Modals/sendModal.tsx [hunk 3]
```diff
@@ -471,6 +473,33 @@ const SendModal = (props: SendModalProps): JSX.Element => {
     setShieldAddress(shieldAddressGet());
   }, []);
 
+  useEffect(() => {
+    if (recipient && recipient.length === 66) {
```
[the remark this answers is not in the document]
Comment (luizoamorim) [reply wallet/src/components/Modals/sendModal.tsx:477]: The function from keys.mjs ?

File: wallet/src/components/BridgeComponent/index.jsx [hunk 4]
```diff
@@ -191,10 +191,17 @@ const BridgeComponent = () => {
...
+            readyTx.rawTransaction,
+            shieldContractAddress,
+            150000,
+            0,
+          ); // 150k is enough gasLimit for a deposit
+          readyTx.transaction.l1Hash = txL1Hash;
           break;
-        case 'offchain':
+        }
+        case 'offchain': {
```
[the remark this answers is not in the document]
Comment (luizoamorim) [reply wallet/src/components/BridgeComponent/index.jsx:204]: Yeah I think was a problem with eslint. I"ll remove and try push.

File: wallet/src/components/BridgeComponent/index.jsx (hunk 1 above)
Comment (luizoamorim) [reply wallet/src/components/BridgeComponent/index.jsx:216]: Yeah I think was a problem with eslint. I"ll remove and try push.

**Comment by luizoamorim** [comment]
The last commits are solving the hardcoded issue @IlyasRidhuan

File: wallet/src/components/Modals/sendModal.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (luizoamorim) [reply wallet/src/components/Modals/sendModal.tsx:477]: Ilyas if you can give me a hand with this function. I'm having some troubles.

File: wallet/src/components/BridgeComponent/index.jsx (hunk 1 above)
Comment (luizoamorim) [reply wallet/src/components/BridgeComponent/index.jsx:216]: If you don't want this brackets ... we'll need to the eslint rule for that.

https://eslint.org/docs/rules/no-case-declarations

File: wallet/src/components/BridgeComponent/index.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (luizoamorim) [reply wallet/src/components/BridgeComponent/index.jsx:204]: If you don't want this brackets ... we'll need to the eslint rule for that.

https://eslint.org/docs/rules/no-case-declarations

**Review by druiz0992 — APPROVED** [review]

**Review by IlyasRidhuan — APPROVED** [review]

**Pull request merged**